sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_concat_placeholder.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Google LLC
```

Comment:
Why can't these be unit tests?

Reply:
A unit test would be on the kfp.components level that we didn't change for v2 compiler, and it will pass on the unit test level but not necessarily on the integration test level. For instance, if-placeholder would work on if we unit test kfp.components but it doesn't work e2e in v2 compiler prior to the other change #4648.

Reply:
>Why can't these be unit tests?

Perhaps this **is** a unit test? See `sdk/python/kfp/v2/compiler_cli_tests/compiler_cli_tests.py`

Reply:
Right, it's a bit tricky here. All these tests are technically unit tests as they are run with unittest framework. But mentally, I would often think a unit test as testing the minimum scope -- a single function or class. While this test involves multiple real (non-mocked) modules, including components, dsl, and compiler, so scope-wise it's more like an integration test, or even e2e test for the sdk.

Reply:
In this case, we could say that this test is an integration test by design. Resolving the `concat` placeholder is the responsibility of the component library and it has a unit test for it. This test just checks it end-to-end on the new compiler, proving that the feature is not somehow broken and is compatible with IR.